### PR TITLE
Feature/add quantity badge on mobile filters button

### PR DESCRIPTION
**Pull request by pedrobernardina**
#### What problem is this solving?

At the search page, where products are shown, the user can apply different filters in order to get better results. On desktop, selected filters are displayed on the left side of the screen, and users always know how many filters they've applied.

On mobile, the filters menu is inside a drawer, that is toggled by clicking a button. However, there is no indication if there's any active filter, forcing users to open the drawer in order to know how many filters they have applied.

We added a prop `showQuantityBadgeOnMobile` that enables a badge on top of the filters button, similar to what we have on MinicartIconButton, indicating how many filters the user has applied. It is set to `false` as default, so it doesn't change the behaviour of other stores.

#### How to test it?

Go to the search page on mobile, and select any filter. The number of current filters will be displayed on top of the button that opens the drawer.

[Workspace](https://adicionabadgenosfiltrosselecionados--oficinareserva.myvtex.com/branco/p/pima?_q=pima&fuzzy=0&initialMap=ft&initialQuery=pima&map=cor,tamanho,ft&operator=and)

#### Screenshots or example usage:

Screenshot from 2022-01-10 14-16-19
Screenshot from 2022-01-10 14-16-33

#### Describe alternatives you've considered, if any.

We read the docs and code of FilterNavigation.v3 to see if there was any way of enabling the desired behaviour, but there is none.

### CODE REVIEW

**Comment by pedrobernardina** [comment]
@victorhmp @igorbrasileiro 

Bom dia! Vocês conseguem dar uma força nesse PR?

@jpesce para visibilidade.

**Review by hiagolcm — APPROVED** [review]
Hey! Thanks for the PR. The feature is nice and the code LGTM

Let's wait for one more review and then we are good to deploy

@thalytafabrine Could you give your opinion?

File: react/components/FilterSidebar.js
Comment (thalytafabrine) [inline react/components/FilterSidebar.js:82]: why do we need to do this? I think `selectedFacets` will always be the same thing as `selectedFilters` 🤔

Comment (pedrobernardina) [reply react/components/FilterSidebar.js:82]: @thalytafabrine thanks for the reply!

In some cases, `selectedFilters` has categories included. In [this page](https://adicionabadgenosfiltrosselecionados--oficinareserva.myvtex.com/roupas/polos/m/preto?initialMap=c,c&initialQuery=roupas/polos&map=category-1,category-2,tamanho,cor), for instance, we have "Roupas" and "Polo" as selected filters. 

This `useMemo` extracts from `selectedFilters` only filters that are shown in the side menu.

Screenshot from 2022-02-16 17-45-04

**Review by Mariana-Caetano — APPROVED** [review]

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]

File: react/components/FilterSidebar.js
Comment (pedrobernardina) [reply react/components/FilterSidebar.js:82]: Hi @igorbrasileiro , thanks for the reviews. 

Here I wanted to return an array of selected filters instead of just a number. I think it's more understandable this way, and allows further uses in the future if necessary.

Do you agree?

File: react/components/FilterSidebar.js
Comment (pedrobernardina) [reply react/components/FilterSidebar.js]: @igorbrasileiro , reviews were done.

@jpesce

File: react/components/FilterSidebar.js
Comment (igorbrasileiro) [reply react/components/FilterSidebar.js:83]: Yeah, i agree, that the selectedFacets array value is good. So, wdyt about returning the `selectedFacets.length` inside the useMemo? By doing this you maintain with the selectedFacets, but only return the interested value, that's the length, if further changes need the selectedFacets the value is already calculated and only the return should change.

Comment (igorbrasileiro) [reply react/components/FilterSidebar.js:83]:
```suggestion
    const filterKeys = filters.map(filter => filter.key);

    return selectedFilters.filter((facet) => (
      filterKeys.includes(facet.key) 
      && facet.selected
    )).length }, [filters, selectedFilters]);
```

File: react/searchResult.css
Comment (igorbrasileiro) [inline react/searchResult.css:239]: Can you explain the motivation of `user-select: none`?! I'm asking about it because it creates a bad UX and it let difficult for people with disabilities to copy this text.

Comment (pedrobernardina) [reply react/searchResult.css:239]: Hi @igorbrasileiro 

I pretty much copied the code and styles from MinicartIconButton, since I wanted to keep the feature as close as possible from components VTEX already have.

You can check the original source on the files below:

https://github.com/vtex-apps/minicart/blob/4ec5e58ca4bd4a30772ed11990023743aadfe15f/react/components/MinicartIconButton.tsx

https://github.com/vtex-apps/minicart/blob/4ec5e58ca4bd4a30772ed11990023743aadfe15f/react/styles.css

But since these icons will appear on mobile only, we don't really need this attribute.

File: react/components/FilterSidebar.js
Comment (pedrobernardina) [reply react/components/FilterSidebar.js:83]: I'd still prefer to return the array, since it took me a while to understand what was going on with `selectedFilters`. An  explicit variable like this would have saved me some time.

But I decided to follow your suggestion, and added a comment on top of the `useMemo`.

File: react/components/FilterSidebar.js
Comment (pedrobernardina) [reply react/components/FilterSidebar.js:223]: Why? :open_mouth:

File: react/searchResult.css
Comment (igorbrasileiro) [reply react/searchResult.css]: Ok, np!

File: react/components/FilterSidebar.js
Comment (igorbrasileiro) [reply react/components/FilterSidebar.js:229]: Sorry was a misunderstanding.

**Review by igorbrasileiro — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]
LGTM

**Pull request merged**
